### PR TITLE
use c++11 std::align from <memory>

### PR DESCRIPTION
**Pull request by psi29a**
* cleans up .travis-ci.yml
* uses named matrix that we can now extend
* clang static analysis done in two parts: openmw and openmw-cs
* replaces old precise clang (3.6) with trusty clang (6.0)
* replaces gcc 4.8 (Trusty default) with gcc 8 (for Trusty)
* replaces inline align with c++11 std::align

### CODE REVIEW

**Comment by psi29a** [comment]
@akortunov @elsid 
This OK for you guys?

**Comment by akortunov** [comment]
It seems the Clang does not have the std::align too. If we need to drop support of older compilers just for one function, maybe we should keep our current implementation?

**Comment by psi29a** [comment]
The compilers have always been available for Trusty and gcc 4.8 and clang-3.6 are not fully C++11 complaint anyway. 

Trusty is EoL in April 2019 anyway.

So we'll be bumping compilers anyway when Travis-CI upgrades to Xenial which is now in Beta.

This is already the case in AppVeyor.

**Comment by psi29a** [comment]
We've used clang's static analyzer forever now, there tons of flags we can set as well. We can investigate those in another PR.

Let's get this squashed and merged.

File: CI/before_script.linux.sh [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh -e
+#!/bin/bash -ex
```
Comment (symbiogenesis) [inline CI/before_script.linux.sh]: did this need to switch to bash specifically? Isn't it better to be agnostic?

Comment (psi29a) [reply CI/before_script.linux.sh]: dash (default sh in ubuntu) doesn't support =~ operator that I needed when testing for clang, so bash it is.

However, if you can do the same and set it bash sh, please send a PR :)

Don't forget, this is for CI builds on Travis-CI and AppVeyor where bash is expected. Linux based releases are not built with these scripts, that is the job of the package managers.

**Pull request merged**